crates/nu-cli/src/completions/completer.rs

```diff
@@ -223,46 +290,13 @@ impl NuCompleter {
 
             // Complete based on the last span
             if is_last_span {
-                // Context variables
-                let most_left_var = most_left_variable(flat_idx, &working_set, flattened.clone());
-
                 // Create a new span
                 let new_span = Span::new(span.start, span.end - 1);
 
                 // Parses the prefix. Completion should look up to the cursor position, not after.
                 let index = pos - span.start;
                 let prefix = &current_span[..index];
 
-                // Variables completion
-                if prefix.starts_with(b"$") || most_left_var.is_some() {
-                    let mut variable_names_completer =
-                        VariableCompletion::new(most_left_var.unwrap_or((vec![], vec![])));
-
-                    let mut variable_completions = self.process_completion(
-                        &mut variable_names_completer,
-                        &working_set,
-                        prefix,
-                        new_span,
-                        fake_offset,
-                        pos,
-                    );
-
-                    let mut variable_operations_completer =
-                        OperatorCompletion::new(element_expression.clone());
-
-                    let mut variable_operations_completions = self.process_completion(
-                        &mut variable_operations_completer,
-                        &working_set,
-                        prefix,
-                        new_span,
-                        fake_offset,
-                        pos,
-                    );
-
-                    variable_completions.append(&mut variable_operations_completions);
```

Comment:
I'm not sure what's this is about, but I moved it without breaking any test. @uek-1, could you please help to review this?

Reply:
Ok I get it, variable operator completion only works with mutable variable, which is not quite accurate, "non-assignment" ops should still work for immutable variables.

And I'm also considering refactoring `OperatorCompletion` and `nu_protocol::ast::Operator` so that the latter contains descriptions, and the prior follows the structure of the `Operator` enum to avoid code duplication, and it is also easier to add new operators.

I'll leave it to another PR.

Reply:
@ysthakur plz double check this

Reply:
Yeah, this looks fine to me. Looks like we don't have any tests for operator completions at the moment, which explains why it wasn't caught. I didn't bother doing a `git bisect` to see where exactly operator completions for variables stopped working, but it was before this PR anyway. So it's best to leave fixing that and the changes to `OperatorCompletion` to another PR, like you said.